File: CPP/Film.cpp
#include "EntitasBioskop.cpp"

class Film : public EntitasBioskop {
protected:
    string judul;
    string genre;
    int durasi;

public:
    Film() : EntitasBioskop() {}
    Film(string id, string nama, int tahunOperasional, string judul, string genre, int durasi)
        : EntitasBioskop(id, nama, tahunOperasional) {
        this->judul = judul;
        this->genre = genre;
        this->durasi = durasi;
    }

    // Getter & Setter
    string getJudul() const { return judul; }
    void setJudul(string judul) { this->judul = judul; }

    string getGenre() const { return genre; }
    void setGenre(string genre) { this->genre = genre; }

    int getDurasi() const { return durasi; }
    void setDurasi(int durasi) { this->durasi = durasi; }
};

File: PHP_TP2/Film.php
<?php
require_once __DIR__ . '/EntitasBioskop.php';

class Film extends EntitasBioskop {
    protected string $judul;
    protected string $genre;
    protected int $durasi;

    public function __construct(string $id, string $nama, int $tahunOperasional, string $judul, string $genre, int $durasi) {
        parent::__construct($id, $nama, $tahunOperasional);
        $this->judul = $judul;
        $this->genre = $genre;
        $this->durasi = $durasi;
    }

    public function getJudul(): string { return $this->judul; }
    public function setJudul(string $judul): void { $this->judul = $judul; }

    public function getGenre(): string { return $this->genre; }
    public function setGenre(string $genre): void { $this->genre = $genre; }

    public function getDurasi(): int { return $this->durasi; }
    public function setDurasi(int $durasi): void { $this->durasi = $durasi; }
}
?>